react-exercise/src/App.tsx

```diff
@@ -3,23 +3,52 @@ import { User } from "./types/user.ts";
 import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
 import { useUsersStore } from "./stores/usersStore.ts";
 import UserProfile from "./pages/userProfile/UserProfile.tsx";
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
+import "./App.css";
+
+// Define a type for fetch state
+type FetchState = "loading" | "success" | "error";
 
 function App() {
   const users = useUsersStore((state) => state.users);
+  const [fetchState, setFetchState] = useState<FetchState>("loading"); // Manage state with a single variable
+  const [errorMessage, setErrorMessage] = useState<string | null>(null); // For error messages
```

Comment:
instead of having the errorMessage separately, you can have it as part of the "FetchState" type

you can have type FetchState = ProgressState | SuccessState | FailureState,

Where: 
type ProgressState = {
type: "progress";
}

type SuccessState = {
type: "success";
}

type FailureState = {
type: "failure";
errorMessage: string;
}

This makes sure that you will have an error message only if there is a failure, and is more type safe